plugins/plugin-chart-pivot-table/src/PivotTableChart.tsx

```diff
@@ -118,16 +90,92 @@ export default function PivotTableChart(props: PivotTableProps) {
       ...acc,
       ...metricNames.map((name: string) => ({
         ...record,
-        metric: name,
+        [METRIC_KEY]: name,
         value: record[name],
       })),
     ],
     [],
   );
 
   const [rows, cols] = transposePivot
-    ? [groupbyColumns, ['metric', ...groupbyRows]]
-    : [groupbyRows, ['metric', ...groupbyColumns]];
+    ? [groupbyColumns, [METRIC_KEY, ...groupbyRows]]
+    : [groupbyRows, [METRIC_KEY, ...groupbyColumns]];
+
+  const handleChange = useCallback(
+    (filters: SelectedFiltersType) => {
+      const groupBy = Object.keys(filters);
+      const groupByValues = Object.values(filters);
+      setDataMask({
+        extraFormData: {
+          filters:
+            groupBy.length === 0
+              ? []
```

Comment:
To avoid unnecessary data requests, it is now (since https://github.com/apache/superset/pull/14408) preferable not to set `extraFormData` values with empty arrays. Instead they should be set to `undefined. This will help the frontend detect that a new chart data request isn't needed. So a small change here:
```suggestion
              ? undefined
```